src/module.c

```diff
@@ -858,9 +858,17 @@ int TSDB_incrby(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     if (RMUtil_ParseArgs(argv, argc, 2, "d", &incrby) != REDISMODULE_OK)
         return RedisModule_WrongArity(ctx);
 
+
+    long long currentUpdatedTime = -1;
+    int timestampLoc = RMUtil_ArgIndex("TIMESTAMP", argv, argc);
+    if (timestampLoc == -1 || RMUtil_StringEqualsC(argv[timestampLoc + 1], "*")) {
```

Comment:
@ashtul @gkorland comment on this code block was correct, his code is easier to understand and will cause less confusion when looking at this parsing code.

Reply:
@danni-m it's ok, I think @ashtul is even cleaner than my